pkg/bwagreement/server_test.go

```diff
@@ -0,0 +1,154 @@
+// Copyright (C) 2018 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package bwagreement
+
+import (
+	"context"
+	"crypto"
+	"crypto/ecdsa"
+	"fmt"
+	"log"
+	"net"
+	"testing"
+
+	"go.uber.org/zap"
+	"google.golang.org/grpc"
+
+	"github.com/stretchr/testify/assert"
+
+	dbx "storj.io/storj/pkg/bwagreement/dbx"
+	"storj.io/storj/pkg/pb"
+	"storj.io/storj/pkg/provider"
+)
+
+var (
+	ctx = context.Background()
+)
+
+const (
+	host     = "localhost"
+	port     = 5432
+	user     = "postgres"
+	password = "your-password"
+	dbname   = "pointerdb"
+)
+
+func getPSQLInfo() string {
+	psqlInfo := fmt.Sprintf("host=%s port=%d user=%s "+
+		"password=%s dbname=%s sslmode=disable",
+		host, port, user, password, dbname)
+	return psqlInfo
+}
+
+func TestBandwidthAgreements(t *testing.T) {
+	TS := NewTestServer(t)
+	defer TS.Stop()
+
+	signature := []byte("iamthedummysignatureoftypebyteslice")
+	data := []byte("iamthedummydataoftypebyteslice")
+
+	msg := &pb.RenterBandwidthAllocation{
+		Signature: signature,
+		Data:      data,
+	}
+
+	/* emulate sending the bwagreement stream from piecestore node */
+	stream, err := TS.c.BandwidthAgreements(ctx)
+	assert.NoError(t, err)
+	err = stream.Send(msg)
+	assert.NoError(t, err)
+
+	_, _ = stream.CloseAndRecv()
+
+	/* read back from the postgres db in bwagreement table */
+	retData, err := TS.s.DB.Get_Bwagreement_By_Signature(ctx, dbx.Bwagreement_Signature(signature))
+	assert.EqualValues(t, retData.Data, data)
+	assert.NoError(t, err)
+
+	/* delete the entry what you just wrote */
+	delBool, err := TS.s.DB.Delete_Bwagreement_By_Signature(ctx, dbx.Bwagreement_Signature(signature))
+	assert.True(t, delBool)
+	assert.NoError(t, err)
+}
+
+type TestServer struct {
+	s     *Server
+	grpcs *grpc.Server
+	conn  *grpc.ClientConn
+	c     pb.BandwidthClient
+	k     crypto.PrivateKey
```

Comment:
Shortening fields makes code more difficult to read for beginners. Use `server`, `client` and `privateKey`.